### PR TITLE
Only print Unexpected Errors

### PR DESCRIPTION
**Pull request by sambitesh**
Earlier ORCA would log all erorrs. Even when they are due to unsupported
features. During gpinit when we are running catalog queries which ORCA doesn't
support by default, we ended up rpinting unecessary error messages to the log.

Now we want to log only unexpected errors. This will cut lot of clutter from log files.

Co-authored-by: Abhijit Subramanya <asubramanya@pivotal.io>
Co-authored-by: Sambitesh Dash <sdash@pivotal.io>

### CODE REVIEW

File: libgpos/src/_api.cpp [hunk 1]
```diff
@@ -22,10 +22,88 @@
 #include "gpos/task/CAutoTaskProxy.h"
 #include "gpos/task/CWorkerPoolManager.h"
 
+
```
Comment (ashuka24) [inline libgpos/src/_api.cpp]: remove extra line

File: libgpos/src/_api.cpp [hunk 2]
```diff
@@ -22,10 +22,88 @@
...
+#include "gpopt/exception.h"
+
 using namespace gpos;
 
+const ULONG expected_opt_fallback[] =
+{
+	gpopt::ExmiInvalidPlanAlternative,		// chosen plan id is outside range of possible plans
+	gpopt::ExmiUnsupportedOp,				// unsupported operator
+	gpopt::ExmiUnsupportedPred,				// unsupported predicate
+	gpopt::ExmiUnsupportedCompositePartKey	// composite partitioning keys
```
Comment (chrishajas) [inline libgpos/src/_api.cpp]: Should `ExmiUnsupportedNonDeterministicUpdate` also be added here?

File: libgpos/CMakeLists.txt [hunk 3]
```diff
@@ -17,6 +17,9 @@ endif()
 
 include_directories(include)
 
+include_directories(${PROJECT_SOURCE_DIR}/libgpopt/include/)
+include_directories(${PROJECT_SOURCE_DIR}/libnaucrates/include/)
```
Comment (chrishajas) [inline libgpos/CMakeLists.txt]: Why is this needed now?

File: libgpos/src/_api.cpp [hunk 4]
```diff
@@ -22,10 +22,88 @@
...
+
+// array of DXL minor exception types that error out and NOT fallback to the planner
+const ULONG expected_dxl_errors[] =
+{
+	gpdxl::ExmiDXL2PlStmtExternalScanError,	// external table error
+	gpdxl::ExmiQuery2DXLNotNullViolation,	// not null violation
+};
+
+BOOL
+ShouldErrorOut
```
Comment (zellerh) [inline libgpos/src/_api.cpp]: Is this method used anywhere?

Comment (sambitesh) [reply libgpos/src/_api.cpp]: It's used in GPDB side.

https://github.com/greenplum-db/gpdb/blob/beac8ab91e44903e67f9ab1207f20ed2716902f7/src/backend/gpopt/utils/COptTasks.cpp#L794

File: libgpos/CMakeLists.txt (hunk 3 above)
Comment (sambitesh) [reply libgpos/CMakeLists.txt]: These two lines are needed to compile correctly since we have included `gpopt/exception.h` and `naucrates/exception.h` in the `_api.cpp` which is inside `gpos` directory.

File: libgpos/src/_api.cpp (hunk 2 above)
Comment (sambitesh) [reply libgpos/src/_api.cpp]: Done.
